Add default tab on downloads page based on user OS

wackerow opened the pull request:
## Description
- Checks the users operating system and sets this as the default tab when navigating to the downloads page.

## Preview link
https://deploy-preview-167--melodious-puffpuff-8e1109.netlify.app/downloads

nhsz commented on src/components/UI/downloads/DownloadsTable.tsx:
```diff
@@ -37,6 +37,15 @@ export const DownloadsTable: FC<Props> = ({
   const LAST_2_LINUX_RELEASES = amountOfReleasesToShow + 12;
 
+  const getDefaultIndex = useMemo<number>(() => {
+    const OS: string = typeof window !== 'undefined' ? window.navigator.platform : '';
```
I think this check doesn't work on mobile browsers

wackerow replied:
Gotcha... okay, it seems to be working on iOS, but I'll look into it

nhsz reviewed (changes requested):
the OS detection doesn't work on mobile, we should [check userAgent instead](https://developer.mozilla.org/en-US/docs/Web/API/Window/navigator)

wackerow commented:
It's working for me on iPhone, I take it Android is not? Can look into it shortly

wackerow commented:
So I don't think we can only use the user agent, since we're looking for the operating system not the browser. I can confirm this works on Linux, Windows, macOS and iOS... if it isn't working on Android, I can add a fallback to also check for `userAgent.includes("SamsungBrowser")` and default to Android in that case.

wackerow commented:
@nhsz I didn't have luck when testing with BrowserStack using just the SamsungBrowser suggested in that mdn link, but checking for "Android" within user agent seems to work. 

Just double checked that making the OS check case-insensitive didn't fix this, which it didn't. Removed the OS check for "Android" and replaced with a user agent check for "Android" instead which seems to do the trick:

<img width="445" alt="image" src="https://user-images.githubusercontent.com/54227730/207983968-b355ac96-370b-4867-bf96-6dffac1f7661.png">

Should be good to merge now.
